src/scripts/main.js

```diff
@@ -1,3 +1,320 @@
+
+  document.addEventListener('keydown', function() {
+    switch (event.key) {
+      case 'ArrowLeft':
+        upDatefield = keyLeft(upDatefield);
+        upDatefield = replaceRandomZero(upDatefield);
+        initial(upDatefield);
+        styleCells();
+        checker();
+        break;
+
+      case 'ArrowRight':
+        upDatefield = keyRight(upDatefield);
+        upDatefield = replaceRandomZero(upDatefield);
+        initial(upDatefield);
+        styleCells();
+        checker();
+        break;
+
+      case 'ArrowUp':
+        upDatefield = keyUp(upDatefield);
+        upDatefield = replaceRandomZero(upDatefield);
+        initial(upDatefield);
+        styleCells();
+        checker();
+        break;
+
+      case 'ArrowDown':
+        upDatefield = keyDown(upDatefield);
+        upDatefield = replaceRandomZero(upDatefield);
+        initial(upDatefield);
+        styleCells();
+        checker();
+        break;
+
+      default:
+    }
+  });
+});
+
+function initial(arr) {
+  for (let i = 0; i < cells.length; i++) {
+    cells[i].innerHTML = arr[i];
+  }
+}
+
+// PRESS LEFT <-----------------------------------
+// Split the flat array into rows
+function keyLeft(field) {
+  let matrix = field;
+  const rows = [];
+
+  for (let i = 0; i < matrix.length; i += 4) {
+    rows.push(matrix.slice(i, i + 4));
+  }
+
+  rows.map((item, index) => {
+    rows[index] = moveLeft(item);
+  });
+
+  matrix = [].concat(...rows);
+
+  // Now `field` is updated based on the manipulated rows
+  function moveLeft(row) {
+    // remove 0's
+    let tempRow = row;
+
+    tempRow = removeZero(row);
+
+    // move & sum rows near cells
+    for (let i = 0; i < tempRow.length - 1; i++) {
+      if (tempRow[i + 1] === tempRow[i]) {
+        scoreNumber += tempRow[i] * 2;
+        [tempRow[i], tempRow[i + 1]] = [tempRow[i] * 2, 0];
+      }
+    }
+    tempRow = removeZero(tempRow);
+
+    tempRow = fillArrayWithZero(tempRow);
+
+    return tempRow;
+  }
+
+  if (JSON.stringify(upDatefield) === JSON.stringify(matrix)) {
+    return upDatefield;
+  }
+
+  upDatefield = matrix;
+  scoreText[0].innerHTML = scoreNumber;
+  // checker();
+
+  return upDatefield;
+}
+
+// PRESS RIGHT ----------------------------------->
+
+function keyRight(field) {
+  // Split the flat array into rows
+  let tempField = field;
+  const rows = [];
+
+  for (let i = 0; i < tempField.length; i += 4) {
+    rows.push(tempField.slice(i, i + 4));
+  }
+
+  rows.map((item, index) => {
+    rows[index] = item.reverse();
+    rows[index] = moveRight(item);
+  });
+
+  tempField = [].concat(...rows);
+
+  function moveRight(row) {
+    let tempRow = row;
+
+    tempRow = removeZero(tempRow);
+
+    // move & sum rows near cells
+    for (let i = 0; i < tempRow.length - 1; i++) {
+      if (tempRow[i + 1] === tempRow[i]) {
+        scoreNumber += tempRow[i] * 2;
+        [tempRow[i], tempRow[i + 1]] = [tempRow[i] * 2, 0];
+      }
+    }
+
```

Comment:
try to simplify slide logic, 'cause you repeat many things in each slide function, try to create something general